--- gtk+-2.20.1/gdk-pixbuf/Makefile.am~	2010-09-22 21:35:35.766910029 +0200
+++ gtk+-2.20.1/gdk-pixbuf/Makefile.am	2010-09-22 21:40:41.460243363 +0200
@@ -692,7 +692,7 @@
 if HAVE_INTROSPECTION
 
 GdkPixbuf-2.0.gir: libgdk_pixbuf-2.0.la Makefile
-GdkPixbuf_2_0_gir_SCANNERFLAGS = --strip-prefix=Gdk
+GdkPixbuf_2_0_gir_SCANNERFLAGS = --identifier-prefix=Gdk --symbol-prefix=gdk
 GdkPixbuf_2_0_gir_INCLUDES = GModule-2.0 Gio-2.0
 GdkPixbuf_2_0_gir_LIBS = libgdk_pixbuf-2.0.la
 GdkPixbuf_2_0_gir_CFLAGS = $(INCLUDES)
--- gtk+-2.20.1/gtk/Makefile.am~	2010-05-03 01:33:18.000000000 +0200
+++ gtk+-2.20.1/gtk/Makefile.am	2010-09-22 21:41:55.953576697 +0200
@@ -982,7 +982,7 @@
     gtktypebuiltins.c
 
 Gtk-2.0.gir: $(INTROSPECTION_SCANNER) $(gtktargetlib) $(top_builddir)/gdk/Gdk-2.0.gir Makefile
-Gtk_2_0_gir_SCANNERFLAGS = --strip-prefix=Gtk
+Gtk_2_0_gir_SCANNERFLAGS = --identifier-prefix=Gtk --symbol-prefix=gtk
 Gtk_2_0_gir_INCLUDES = Atk-1.0 Gdk-2.0
 Gtk_2_0_gir_CFLAGS = \
 		$(INCLUDES) \
--- gtk+-2.20.1/gdk/Makefile.am~	2010-05-03 01:33:18.000000000 +0200
+++ gtk+-2.20.1/gdk/Makefile.am	2010-09-22 21:46:40.213576696 +0200
@@ -202,7 +202,7 @@
         $(srcdir)/x11/*.c
 
 Gdk-2.0.gir: $(gdktargetlib) Makefile
-Gdk_2_0_gir_SCANNERFLAGS = --strip-prefix=Gdk --add-include-path=../gdk-pixbuf
+Gdk_2_0_gir_SCANNERFLAGS = --identifier-prefix=Gdk --symbol-prefix=gdk --add-include-path=../gdk-pixbuf
 Gdk_2_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0
 Gdk_2_0_gir_LIBS = $(gdktargetlib)
 Gdk_2_0_gir_FILES = $(addprefix $(srcdir)/,$(introspection_files))
