diff -urN gtk+-2.22.0/gdk/Makefile.am gtk+-2.22.0.new//gdk/Makefile.am
--- gtk+-2.22.0/gdk/Makefile.am	2010-09-23 05:00:45.000000000 +0200
+++ gtk+-2.22.0.new//gdk/Makefile.am	2010-09-27 20:45:46.955178923 +0200
@@ -238,7 +238,7 @@
         x11/gdkx.h
 
 GdkX11-2.0.gir: $(gdktargetlib) Gdk-2.0.gir Makefile
-GdkX11_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk
+GdkX11_2_0_gir_SCANNERFLAGS = --warn-all --identifier-prefix=Gdk --symbol-prefix=gdk
 GdkX11_2_0_gir_INCLUDES = Gio-2.0 Gdk-2.0 GdkPixbuf-2.0 Pango-1.0 xlib-2.0
 GdkX11_2_0_gir_LIBS = $(gdktargetlib)
 GdkX11_2_0_gir_FILES = $(x11_introspection_files)
