--- gtk+-2.24.23/modules/engines/pixbuf/pixbuf-rc-style.c.orig	2014-03-06 05:56:42.000000000 +0100
+++ gtk+-2.24.23/modules/engines/pixbuf/pixbuf-rc-style.c	2014-03-22 09:12:02.069004277 +0100
@@ -669,7 +669,7 @@
                              const char   *message)
 {
   theme_clear_pixbuf (theme_pb);
-  g_scanner_warn (scanner, message);
+  g_scanner_warn (scanner, "%s", message);
 }
 
 static guint
