--- gtk/gtktreemodel.c	17 Dec 2003 17:52:47 -0000	1.107
+++ gtk/gtktreemodel.c	14 Jan 2004 03:54:18 -0000
@@ -1400,9 +1400,9 @@
  * @tree_model: A #GtkTreeModel
  * @path: A #GtkTreePath pointing to the tree node whose children have been reordered
  * @iter: A valid #GtkTreeIter pointing to the node whose children have been reordered
- * @new_order: an array of integers containing the new indices of the children,
- *      i.e. the former child <literal>n</literal> is now at position
- *      @new_order<literal>[n]</literal>. 
+ * @new_order: an array of integers mapping the current position of each child
+ *      to its old position before the re-ordering,
+ *      i.e. @new_order<literal>[newpos] = oldpos</literal>.
  * 
  * Emits the "rows_reordered" signal on @tree_model.  This should be called by
  * models when their rows have been reordered.  
