Index: gtk/gtktextlayout.c
===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtktextlayout.c,v
retrieving revision 1.83
retrieving revision 1.83.2.1
diff -u -r1.83 -r1.83.2.1
--- gtk/gtktextlayout.c	2002/03/23 22:14:40	1.83
+++ gtk/gtktextlayout.c	2002/04/30 20:51:26	1.83.2.1
@@ -719,11 +719,10 @@
     {
       GtkTextLineData *line_data = _gtk_text_line_get_data (line, layout);
 
+      gtk_text_layout_invalidate_cache (layout, line);
+      
       if (line_data)
-        {
-          gtk_text_layout_invalidate_cache (layout, line);
-          _gtk_text_line_invalidate_wrap (line, line_data);
-        }
+        _gtk_text_line_invalidate_wrap (line, line_data);
 
       if (line == last_line)
         break;
@@ -739,12 +738,7 @@
                                      GtkTextLine       *line,
                                      GtkTextLineData   *line_data)
 {
-  if (layout->one_display_cache && line == layout->one_display_cache->line)
-    {
-      GtkTextLineDisplay *tmp_display = layout->one_display_cache;
-      layout->one_display_cache = NULL;
-      gtk_text_layout_free_line_display (layout, tmp_display);
-    }
+  gtk_text_layout_invalidate_cache (layout, line);
 
   g_free (line_data);
 }
