diff -Nuard gtk+-2.2.4.orig/gtk/gtksocket.c gtk+-2.2.4/gtk/gtksocket.c
--- gtk+-2.2.4.orig/gtk/gtksocket.c	2003-08-25 23:14:22.000000000 +0200
+++ gtk+-2.2.4/gtk/gtksocket.c	2003-09-17 14:34:02.000000000 +0200
@@ -840,7 +840,7 @@
   if (socket->plug_widget)
     return gtk_widget_child_focus (socket->plug_widget, direction);
 
-  if (!GTK_WIDGET_HAS_FOCUS (widget))
+  if (!gtk_widget_is_focus (widget))
     {
       switch (direction)
 	{
