--- gtk+-2.24.21/modules/input/Makefile.am.orig	2013-07-30 02:54:43.000000000 +0200
+++ gtk+-2.24.21/modules/input/Makefile.am	2013-09-18 17:28:45.876189834 +0200
@@ -167,7 +167,7 @@
 endif
 endif
 
-multipress_defs = -DMULTIPRESS_LOCALEDIR=\""$(mplocaledir)"\" -DMULTIPRESS_CONFDIR=\""$(sysconfdir)/gtk-2.0"\"
+multipress_defs = -DMULTIPRESS_LOCALEDIR=\""$(mplocaledir)"\" -DMULTIPRESS_CONFDIR=\""$(sysconfdir)"\"
 im_multipress_la_CPPFLAGS = $(multipress_defs)
 libstatic_im_multipress_la_CPPFLAGS = $(im_multipress_la_CPPFLAGS)
 im_multipress_la_LDFLAGS = -rpath $(moduledir) -avoid-version -module $(no_undefined)
@@ -183,7 +183,7 @@
 MULTIPRESS_MODULE = im-multipress.la
 endif
 
-imconffiledir = $(sysconfdir)/gtk-2.0
+imconffiledir = $(sysconfdir)
 dist_imconffile_DATA = im-multipress.conf
 
 if CROSS_COMPILING
@@ -200,10 +200,10 @@
 # after the libraries are installed in their final locations.
 install-data-hook:
 	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
-	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  echo "$(DESTDOR)$(bindir)/gtk-query-immodules-2.0 > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache" ; \
-	  $(DESTDIR)/$(bindir)/gtk-query-immodules-2.0 > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache ; \
+	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir) ; \
+	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir) ; \
+	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache" ; \
+	  $(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(libdir)/gtk-2.0/$(GTK_BINARY_VERSION)/immodules.cache ; \
 	else \
 	  echo "***" ; \
 	  echo "*** Warning: immodules.cache not built" ; \
--- gtk+-2.24.20/gtk/gtkrc.c.orig	2013-07-07 14:53:53.881925617 +0200
+++ gtk+-2.24.20/gtk/gtkrc.c	2013-07-07 14:57:44.148582620 +0200
@@ -533,7 +533,7 @@
           g_free (str);
         }
 
-      str = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtkrc", NULL);
+      str = g_build_filename (GTK_SYSCONFDIR, "gtkrc", NULL);
       gtk_rc_add_default_file (str);
       g_free (str);
 
