--- gtk+-2.4.3/configure.in~	2004-07-09 19:10:52.890349944 +0000
+++ gtk+-2.4.3/configure.in	2004-07-09 19:14:15.319576024 +0000
@@ -331,7 +331,8 @@
 if test "$os_win32" != yes; then
     # libtool option to control which symbols are exported
     # right now, symbols starting with _ are not exported
-    LIBTOOL_EXPORT_OPTIONS='-export-symbols-regex "^[[^_]].*"'
+    # Disabled until -export-symbols-regex works (RH patch)
+    LIBTOOL_EXPORT_OPTIONS=
 else
     # We currently use .def files on Windows (for gdk-pixbuf, gdk and gtk)
     LIBTOOL_EXPORT_OPTIONS=
