diff -urN gtk+-2.11.6/acconfig.h gtk+-2.11.6.new/acconfig.h
--- gtk+-2.11.6/acconfig.h	2007-07-20 22:18:40.000000000 +0200
+++ gtk+-2.11.6.new/acconfig.h	2007-08-28 22:14:32.000000000 +0200
@@ -72,6 +72,8 @@
 
 #undef HAVE_CUPS_API_1_2
 
+#undef HAVE_CUPS_API_1_3
+
 /* #undef PACKAGE */
 /* #undef VERSION */
 
diff -urN gtk+-2.11.6/configure.in gtk+-2.11.6.new/configure.in
--- gtk+-2.11.6/configure.in	2007-07-20 22:18:40.000000000 +0200
+++ gtk+-2.11.6.new/configure.in	2007-08-28 22:14:00.000000000 +0200
@@ -1565,6 +1565,11 @@
     AC_DEFINE(HAVE_CUPS_API_1_2)
   fi
 
+  if test $CUPS_API_MAJOR -gt 1 -o \
+	  $CUPS_API_MAJOR -eq 1 -a $CUPS_API_MINOR -ge 3; then
+    AC_DEFINE(HAVE_CUPS_API_1_3)
+  fi
+
   AC_SUBST(CUPS_API_MAJOR)
   AC_SUBST(CUPS_API_MINOR)
   AC_SUBST(CUPS_CFLAGS)
diff -urN gtk+-2.11.6/modules/printbackends/cups/gtkcupsutils.c gtk+-2.11.6.new/modules/printbackends/cups/gtkcupsutils.c
--- gtk+-2.11.6/modules/printbackends/cups/gtkcupsutils.c	2007-07-20 22:14:11.000000000 +0200
+++ gtk+-2.11.6.new/modules/printbackends/cups/gtkcupsutils.c	2007-08-28 21:58:52.000000000 +0200
@@ -634,7 +634,9 @@
   httpClearFields (request->http);
   httpSetField (request->http, HTTP_FIELD_CONTENT_LENGTH, length);
   httpSetField (request->http, HTTP_FIELD_CONTENT_TYPE, "application/ipp");
-#ifdef HAVE_HTTP_AUTHSTRING
+#ifdef HAVE_CUPS_API_1_3
+  httpSetField (request->http, HTTP_FIELD_AUTHORIZATION, httpGetAuthString (request->http));
+#else
   httpSetField (request->http, HTTP_FIELD_AUTHORIZATION, request->http->authstring);
 #endif
 
@@ -973,7 +975,9 @@
     }
 
   httpClearFields (request->http);
-#ifdef HAVE_HTTP_AUTHSTRING
+#ifdef HAVE_CUPS_API_1_3
+  httpSetField (request->http, HTTP_FIELD_AUTHORIZATION, httpGetAuthString (request->http));
+#else
   httpSetField (request->http, HTTP_FIELD_AUTHORIZATION, request->http->authstring);
 #endif
 
