--- gtk+-2.8.12/gtk/gtktreeview.c.orig	2005-12-14 19:14:45.000000000 +0100
+++ gtk+-2.8.12/gtk/gtktreeview.c	2006-02-20 06:51:08.000000000 +0100
@@ -9606,11 +9606,11 @@
 
   g_object_notify (G_OBJECT (tree_view), "model");
 
-  if (tree_view->priv->selection)
-  _gtk_tree_selection_emit_changed (tree_view->priv->selection);
-
-  if (GTK_WIDGET_REALIZED (tree_view))
+  if (GTK_WIDGET_REALIZED (tree_view)) {
+    if (tree_view->priv->selection)
+      _gtk_tree_selection_emit_changed (tree_view->priv->selection);
     gtk_widget_queue_resize (GTK_WIDGET (tree_view));
+  }
 }
 
 /**
