diff -urN gtk+-2.2.1.orig/gtk/gtktreeview.c gtk+-2.2.1/gtk/gtktreeview.c
--- gtk+-2.2.1.orig/gtk/gtktreeview.c	Tue Jan 28 00:18:53 2003
+++ gtk+-2.2.1/gtk/gtktreeview.c	Sun Jun  1 00:19:12 2003
@@ -349,6 +349,7 @@
 static gboolean gtk_tree_view_has_special_cell               (GtkTreeView       *tree_view);
 
 static gboolean expand_collapse_timeout                      (gpointer           data);
+static gboolean do_expand_collapse                           (GtkTreeView       *tree_view);
 
 /* interactive search */
 static void     gtk_tree_view_search_dialog_destroy     (GtkWidget        *search_dialog,
@@ -6297,7 +6298,7 @@
 cancel_arrow_animation (GtkTreeView *tree_view)
 {
   if (tree_view->priv->expand_collapse_timeout)
-    while (expand_collapse_timeout (tree_view));
+    while (do_expand_collapse (tree_view));
   tree_view->priv->expand_collapse_timeout = 0;
 }
 
@@ -8957,14 +8958,25 @@
 static gboolean
 expand_collapse_timeout (gpointer data)
 {
-  GtkTreeView *tree_view = data;
+  gboolean retval;
+
+  GDK_THREADS_ENTER ();
+
+  retval = do_expand_collapse (data);
+
+  GDK_THREADS_LEAVE ();
+
+  return retval;
+}
+
+static gboolean
+do_expand_collapse (GtkTreeView *tree_view)
+{
   GtkRBNode *node;
   GtkRBTree *tree;
   gboolean expanding;
   gboolean redraw;
 
-  GDK_THREADS_ENTER ();
-
   redraw = FALSE;
   expanding = TRUE;
 
@@ -9013,13 +9025,9 @@
     {
       gtk_tree_view_queue_draw_arrow (tree_view, tree, node, NULL);
 
-      GDK_THREADS_LEAVE ();
-
       return TRUE;
     }
 
-  GDK_THREADS_LEAVE ();
-
   return FALSE;
 }
 
