Index: gtk/gtkselection.c
===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkselection.c,v
retrieving revision 1.108
diff -u -r1.108 gtkselection.c
--- gtk/gtkselection.c	7 Mar 2006 13:45:55 -0000	1.108
+++ gtk/gtkselection.c	13 Jun 2006 07:48:01 -0000
@@ -1700,6 +1700,9 @@
 
   /* Keep in sync with gtk_target_list_add_text_targets()
    */
+
+  init_atoms();
+
   for (i = 0; i < n_targets; i++)
     {
       if (targets[i] == utf8_atom ||
@@ -1744,6 +1747,8 @@
 
   g_return_val_if_fail (GTK_IS_TEXT_BUFFER (buffer), FALSE);
 
+  init_atoms();
+
   rich_targets = gtk_text_buffer_get_deserialize_formats (buffer,
                                                           &n_rich_targets);
 
@@ -1930,6 +1935,9 @@
 
   /* Keep in sync with gtk_target_list_add_uri_targets()
    */
+
+  init_atoms();
+
   for (i = 0; i < n_targets; i++)
     {
       if (targets[i] == text_uri_list_atom)
