diff -urN gtk+-2.2.1.orig/gtk/gtktreeview.c gtk+-2.2.1/gtk/gtktreeview.c
--- gtk+-2.2.1.orig/gtk/gtktreeview.c	Thu Mar  6 11:45:48 2003
+++ gtk+-2.2.1/gtk/gtktreeview.c	Thu Mar  6 11:47:36 2003
@@ -4166,8 +4166,7 @@
       need_redraw = TRUE;
     }
 
-  if (tree_view->priv->scroll_to_path &&
-      ! GTK_RBNODE_FLAG_SET (tree_view->priv->tree->root, GTK_RBNODE_DESCENDANTS_INVALID))
+  if (tree_view->priv->scroll_to_path)
     {
       gtk_tree_row_reference_free (tree_view->priv->scroll_to_path);
       tree_view->priv->scroll_to_path = NULL;
