--- gtk+-2.8.12/gtk/gtkrc.c	2005-11-23 15:24:51.000000000 +0100
+++ gtk+-2.8.12.arch_confdir/gtk/gtkrc.c	2006-02-20 22:14:36.000000000 +0100
@@ -364,7 +364,7 @@
       if (im_module_file)
 	result = g_strdup (im_module_file);
       else
-	result = g_build_filename (GTK_SYSCONFDIR, "gtk-2.0", "gtk.immodules", NULL);
+	result = g_build_filename (GTK_SYSCONFDIR, "gtk.immodules", NULL);
     }
 
   return result;
#--- gtk+-2.8.12/gtk-zip.sh.in	2006-02-07 23:57:06.000000000 +0100
#+++ gtk+-2.8.12.arch_confdir/gtk-zip.sh.in	2006-02-20 22:12:22.000000000 +0100
#@@ -9,9 +9,9 @@
# 
# zip $ZIP -@ <<EOF
# COPYING.LIB-2
#-etc/gtk-2.0/gdk-pixbuf.loaders
#-etc/gtk-2.0/gtkrc
#-etc/gtk-2.0/gtk.immodules
#+etc/gtk-2.0-@ARCH@/gdk-pixbuf.loaders
#+etc/gtk-2.0-@ARCH@/gtkrc
#+etc/gtk-2.0-@ARCH@/gtk.immodules
# bin/libgdk_pixbuf-@GTK_API_VERSION@-@LT_CURRENT_MINUS_AGE@.dll
# bin/libgdk-win32-@GTK_API_VERSION@-@LT_CURRENT_MINUS_AGE@.dll
# bin/libgtk-win32-@GTK_API_VERSION@-@LT_CURRENT_MINUS_AGE@.dll
--- gtk+-2.8.12/modules/input/Makefile.am	2005-06-05 00:08:50.000000000 +0200
+++ gtk+-2.8.12.arch_confdir/modules/input/Makefile.am	2006-02-20 22:14:07.000000000 +0100
@@ -101,10 +102,10 @@
 # after the libraries are installed in their final locations.
 install-data-hook:
 	@if $(RUN_QUERY_IMMODULES_TEST) ; then \
-	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir)/gtk-2.0 ; \
-	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules" ; \
-	  $(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules ; \
+	  echo $(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
+	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
+	  echo "$(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk.immodules" ; \
+	  $(top_builddir)/gtk/gtk-query-immodules-2.0 > $(DESTDIR)$(sysconfdir)/gtk.immodules ; \
 	else \
 	  echo "***" ; \
 	  echo "*** Warning: gtk.immodules not built" ; \
@@ -115,7 +116,7 @@
         fi
 
 uninstall-local:
-	rm -f $(DESTDIR)$(sysconfdir)/gtk-2.0/gtk.immodules
+	rm -f $(DESTDIR)$(sysconfdir)/gtk.immodules
 
 module_LTLIBRARIES =				\
 	$(IM_XIM_MODULE)			\
--- gtk+-2.12.9/modules/input/Makefile.am~	2008-03-22 00:23:37.000000000 +0200
+++ gtk+-2.12.9/modules/input/Makefile.am	2008-03-22 00:57:31.244698955 +0200
@@ -91,7 +91,7 @@
 IM_IME_MODULE=im-ime.la 
 endif
 
-multipress_defs = -DMULTIPRESS_LOCALEDIR=\""$(mplocaledir)"\" -DMULTIPRESS_CONFDIR=\""$(sysconfdir)/gtk-2.0"\"
+multipress_defs = -DMULTIPRESS_LOCALEDIR=\""$(mplocaledir)"\" -DMULTIPRESS_CONFDIR=\""$(sysconfdir)"\"
 im_multipress_la_CPPFLAGS = $(multipress_defs)
 im_multipress_la_LDFLAGS = -rpath $(moduledir) -avoid-version -module $(no_undefined)
 im_multipress_la_SOURCES =	\
@@ -100,7 +100,7 @@
 	immultipress.c
 im_multipress_la_LIBADD = $(LDADDS)
 
-imconffiledir = $(sysconfdir)/gtk-2.0
+imconffiledir = $(sysconfdir)
 dist_imconffile_DATA = im-multipress.conf
 
 if CROSS_COMPILING
