diff -Nuard gtk+-2.2.4.orig/gtk/gtkplug.c gtk+-2.2.4/gtk/gtkplug.c
--- gtk+-2.2.4.orig/gtk/gtkplug.c	2003-08-25 23:14:22.000000000 +0200
+++ gtk+-2.2.4/gtk/gtkplug.c	2003-10-29 21:29:02.000000000 +0100
@@ -934,7 +934,7 @@
 
   Atom xembed_info_atom = gdk_x11_get_xatom_by_name_for_display (display, "_XEMBED_INFO");
 
-  buffer[1] = GTK_XEMBED_PROTOCOL_VERSION;
+  buffer[0] = GTK_XEMBED_PROTOCOL_VERSION;
   buffer[1] = flags;
 
   XChangeProperty (GDK_DISPLAY_XDISPLAY (display),
